testing/unittests/jlibtests.cpp

```diff
@@ -3396,7 +3396,416 @@ element1: !el scalarvalue
+                - name: rule-set
+                  value: 'rule-set-2'
+              valueType:
+              - name: secret
+                presence: r
+              - name: secret-if-a
+                presence: p
+                maskStyle:
+                - name: keep-last-4-numbers
+                  presence: r
+                - name: mask-last-4-numbers
+                  presence: o
+                Set:
+                - name: value-type-set-a
+                  presence: r
+              - name: secret-if-b
+                presence: r
+                Set:
+                - name: value-type-set-b
+                  presence: r
+              rule:
+              - contentType: ''
+                presence: r
+              - contentType: xml
+                presence: r
+          - compatibility:
+              context:
+                domain: 'urn:hpcc:unittest'
+                version: 0
+                property:
+                - name: valuetype-set
+                - name: rule-set
+                  value: 'rule-set-2'
+              valueType:
+              - name: secret
+                presence: r
+              - name: secret-if-a
+                presence: p
+                maskStyle:
+                - name: keep-last-4-numbers
+                  presence: r
+                - name: mask-last-4-numbers
+                  presence: o
+                Set:
+                - name: value-type-set-a
+                  presence: r
+              - name: secret-if-b
+                presence: p
+                Set:
+                - name: value-type-set-b
+                  presence: r
+              rule:
+              - contentType: ''
+                presence: p
+              - contentType: xml
+                presence: p
+      )!!!";
+    Owned<IPropertyTree> originalTree;
+
+public:
+    void testSerializeVsSerializeToStreamRootOnlyPTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createPTree("EmptyRoot"));
+    }
+
+    void testSerializeVsSerializeToStreamFromXmlPTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createPTree(testXml));
+    }
+
+    void testSerializeVsSerializeToStreamFromYamlPTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createPTreeFromYAMLString(chkTxt));
+    }
+
+    void testSerializeVsSerializeToStreamCPTValuePTree()
+    {
+        testSerializeVsSerializeToStream(__func__, createCPTValuePTree());
+    }
+
+protected:
+    void testSerializeVsSerializeToStream(const char* testName, Owned<IPropertyTree> _originalTree)
+    {
+        originalTree.setown(_originalTree.getClear());
+
+        DBGLOG("Starting timing test for %s", testName);
+
+        // Time serialize() method
+        CCycleTimer timer;
+        MemoryBuffer memBufOriginalTree;
+        originalTree->serialize(memBufOriginalTree);
+        __uint64 serializedElapsedNs = timer.elapsedNs();
+
+        // Time serializeToStream() method
+        timer.reset();
```

Comment:
pedantic: probably should move just before serializeToStream call really, otherwise timing includes, admittedly insignificant time of MemoryBuffer ctor + call to createBufferedSerialOutputStream.
Similar above on line 3734